libbeat/common/mapstr.go

```diff
@@ -322,6 +322,17 @@ func walkMap(key string, data MapStr, op mapStrOperation) (interface{}, error) {
 	var err error
 	keyParts := strings.Split(key, ".")
 
+	// Only check for full key if key contains dots
+	if len(keyParts) > 1 {
+		// Converted dotted key in object to nested object
+		// This is needed for processors to support dotted keys
```

Comment:
Can you add a test case to the MapStr tests for this. Is this needed for data that has not been de-dotted (the key names contain dots)?